src/main/java/com/fasterxml/jackson/core/JsonFactory.java

```diff
@@ -2138,7 +2160,12 @@ protected JsonGenerator _decorate(JsonGenerator g) {
      */
     public BufferRecycler _getBufferRecycler()
     {
-        return _getBufferRecyclerPool().acquireBufferRecycler(this);
+        return _getBufferRecyclerPool().acquireBufferRecycler();
+    }
+
+    public static BufferRecycler _getDefaultBufferRecycler()
```

Comment:
why is this needed? For me, the method above should suffice.

Reply:
Yeah I don't know why we'd want that.

Reply:
Looks like it is used from IOContext legacy constructor. I'll think of an alternative as this seems bit off.